examples/DotNetCore/WebDemoWithEventHub/WebDemoWithEventHub/EventHubService.cs

```diff
@@ -0,0 +1,107 @@
+                           kv.SetCredential(new DefaultAzureCredential());
+                       })
+                       .Select(keyFilter: "WebDemo:*")
+                       .ConfigureRefresh((refreshOptions) =>
+                       {
+                            refreshOptions.Register(key: "WebDemo:Settings", refreshAll: true);
+                       });
+
+                refresher = options.GetRefresher();
+            });
+
+            configuration = builder.Build();
+
+            eventHubConnection = configuration.GetSection("WebDemo:Connection").Get<EventHubConnection>();
+            settings = configuration.GetSection("WebDemo:Settings").Get<Settings>();
+
+            InitEventHubProcessor();
+        }
+
+        public Settings GetSettings() => settings;
+
+        private void InitEventHubProcessor()
+        {
+            if (string.IsNullOrEmpty(eventHubConnection.EventHubConnectionString))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnection.EventHubConnectionString));
+            }
+
+            if (string.IsNullOrEmpty(eventHubConnection.EventHubName))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnection.EventHubName));
+            }
+
+            if (string.IsNullOrEmpty(eventHubConnection.StorageConnectionString))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnection.StorageConnectionString));
+            }
+
+            if (string.IsNullOrEmpty(eventHubConnection.StorageContainerName))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnection.StorageContainerName));
+            }
+
+            string consumerGroup = string.IsNullOrEmpty(eventHubConnection.EventHubConsumerGroup)
+                ? EventHubConsumerClient.DefaultConsumerGroupName
+                : eventHubConnection.EventHubConsumerGroup;
+
+            var storageClient = new BlobContainerClient(connectionString: eventHubConnection.StorageConnectionString,
+                                                        blobContainerName: eventHubConnection.StorageContainerName);
+
+            processorClient = new EventProcessorClient(checkpointStore: storageClient,
+                                                       consumerGroup: consumerGroup,
+                                                       connectionString: eventHubConnection.EventHubConnectionString,
+                                                       eventHubName: eventHubConnection.EventHubName);
+
+            processorClient.ProcessEventAsync += ProcessEventHandler;
+            processorClient.ProcessErrorAsync += ProcessErrorHandler;
+            processorClient.StartProcessing();
+        }
+
+        private async Task ProcessEventHandler(ProcessEventArgs eventArgs)
+        {
+            await refresher.TryRefreshAsync();
+            settings = configuration.GetSection("WebDemo:Settings").Get<Settings>();
```

Comment:
We already do this in startup.cs. You don't need to do this again here.

Reply:
Removed it.

---
In reply to: [426104072](https://github.com/Azure/AppConfiguration/pull/329#discussion_r426104072) [](ancestors = 426104072)